src/main/java/sparksoniq/jsoniq/runtime/iterator/control/SwitchRuntimeIterator.java

```diff
@@ -89,38 +89,38 @@ private void initializeIterator(
             Map<RuntimeIterator, RuntimeIterator> cases,
             RuntimeIterator defaultReturn
     ) {
-        Item testValue = getSingleItemFromIterator(
-            test
-        );
-
-        if (testValue != null && testValue.isArray()) {
-            throw new NonAtomicKeyException(
-                    "Invalid args. Switch condition can't be an array type",
-                    getMetadata().getExpressionMetadata()
-            );
-        } else if (testValue != null && testValue.isObject()) {
-            throw new NonAtomicKeyException(
-                    "Invalid args. Switch condition  can't be an object type",
-                    getMetadata().getExpressionMetadata()
-            );
-        }
-
-        for (RuntimeIterator caseKey : cases.keySet()) {
-            Item caseValue = getSingleItemFromIterator(
-                caseKey
-            );
+        Item testValue = test.materializeFirstItemOrNull(_currentDynamicContextForLocalExecution);
```

Comment:
Could we check that 

switch((1, 2)) case 1 return 0 default return 2

returns an error?

In other words, do we check there are no extra items?

In a nutshell, we might need to have two rather than one methods depending on whether we check or not:

materializeOneItemOrNull (checks that there is no second one, i.e., hasNext = false after that)

materializeFirstItemOrNull (does NOT check that there is no second one, if we don't care)

Reply:
TODO for later when singleton method is in place